### PR TITLE
Reland "Add test for empty tensors for batch matmuls"

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This reverts commit c7746adbc6e6ace9d4c2b54e32c8d36a7b7b0e31.

Fixes #{issue number}

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
@mruberry @ngimel This is working

**Review by mruberry — APPROVED** [review]
Cool!

What changed that pytorch_windows_vs2019_py36_cuda11.1_test2 is no longer breaking?

**Comment by zasdfgbnm** [comment]
@mruberry It is https://github.com/pytorch/pytorch/pull/48797/commits/9ae70284c75e82bda368352a148a77091a6742e3

File: test/test_linalg.py [hunk 1]
```diff
@@ -3632,18 +3632,31 @@ def invert_perm(p):
...
                 yield b1, b2
+            # broadcasting tensors
             for b1, b2, b3, b4, b5, b6 in itertools.product((True, False), repeat=6):
                 shape1 = (num_batches if b1 else 1, M if b2 else 1, N if b3 else 1)
                 shape2 = (num_batches if b4 else 1, N if b5 else 1, O if b6 else 1)
                 b1 = make_tensor(shape1, device, dtype, low=-1, high=1).expand(num_batches, M, N)
                 b2 = make_tensor(shape2, device, dtype, low=-1, high=1).expand(num_batches, N, O)
                 yield b1, b2
+            # zero-sized tensors
+            bug = (self.device_type == 'cuda' and dtype == torch.half and torch.version.cuda is not None and
```
Comment (ngimel) [inline test/test_linalg.py:3650]: wait, does cublas < 11 not handle empty tensors correctly? That's interesting. DO we have an issue for that to cite here?

Comment (zasdfgbnm) [reply test/test_linalg.py:3650]: It was, but I don't think it is anymore after https://github.com/pytorch/pytorch/commit/9ae70284c75e82bda368352a148a77091a6742e3, I will update it.

Comment (zasdfgbnm) [reply test/test_linalg.py]: tests are passing, seem to be OK

**Comment by zasdfgbnm** [comment]
ping @ngimel

**Pull request closed without merging**